### PR TITLE
Make testing more robust

### PR DESCRIPTION
**Pull request by billschereriii**
Make sure to shut down Redis server even if tests fail. 
Do so while still reporting the result of testing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -320,9 +320,11 @@ define run_smartredis_tests_with_standalone_server
 	echo "Running standalone tests" && \
 	PYTHONFAULTHANDLER=1 python -m pytest $(SR_TEST_PYTEST_FLAGS) $(COV_FLAGS) \
 		$(SKIP_DOCKER) $(SKIP_PYTHON) $(SKIP_FORTRAN) \
-		--build $(SR_BUILD) --sr_fortran $(SR_FORTRAN) $(1) && \
+		--build $(SR_BUILD) --sr_fortran $(SR_FORTRAN) $(1)  ; \
+	(testresult=$$?; \
 	echo "Shutting down standalone Redis server" && \
 	python utils/launch_redis.py --port $(SR_TEST_PORT) --nodes 1 --stop && \
+	exit $$testresult) && \
```
Comment (ashao) [inline Makefile:327]: I think this would be slightly more bash-like:
```
test $$testresult -eq 0 || echo "Tests failed"; exit $$test_result
```

**Review by ashao — CHANGES_REQUESTED** [review]
Small tweak, but I want to make sure that you agree from a readability/correctness standpoint

File: Makefile (hunk 1 above)
Comment (billschereriii) [reply Makefile:327]: I like that, will adopt

**Review by ashao — APPROVED** [review]
Looks great thanks Bill!

**Review by MattToast — APPROVED** [review]
LGTM!!

**Pull request merged**